### PR TITLE
SelectPanel2: Make room for custom elements in Header

### PR DESCRIPTION
**Pull request by siddharthkp**
- Closes https://github.com/github/primer/issues/2613
- Implements ideas tested in https://github.com/primer/react/pull/3731
- Part of https://github.com/github/primer/issues/2396
- Accessibility guidance for "Tabs" / Filter buttons: https://github.com/github/accessibility/issues/4156#issuecomment-1701473661

Decision:

Instead of creating a `SelectPanel.Tabs` component, we are opening up some space in `SelectPanel.Header` for ItemPicker to utilise. The promise here is that ItemPicker would be the first (and maybe only) user of drafts/SelectPanel for a while. This way, we can learn from usage in dotcom and see if we want to bake some decisions / add constraints to this open slot in `SelectPanel.Header`

API:

```diff
<SelectPanel>

  <SelectPanel.Header>
    <SelectPanel.SearchInput/>

+   {/*  children without slots in header */}
+   <Box sx={{display:'flex'}}>
+     <Button variant="invisible">Branches</Button>
+     <Button variant="invisible">Tags</Button>
+   </Box>
+   {/*  end open children */}
  </SelectPanel.Header>

  {/*  list, actions, etc */}

</SelectPanel>
```

### CODE REVIEW

File: src/drafts/SelectPanel2/SelectPanel.stories.tsx [hunk 1]
```diff
@@ -471,6 +472,149 @@ export const TODO3NoCustomisation = () => {
...
+                  </>
+                )
+              })}
+            </>
+          )}
+        </SelectPanel.ActionList>
+        <SelectPanel.Footer>
+          {/* TODO: Can't disable Cancel and Save yet */}
+          <SelectPanel.SecondaryButton as="a" href="/branches">
+            See all branches
```
Comment (gr2m) [inline src/drafts/SelectPanel2/SelectPanel.stories.tsx:610]: In this particular example, we want to show a different secondary link depending on the selected tab ("View all branches" or "View all tags")

File: src/drafts/SelectPanel2/SelectPanel.stories.tsx [hunk 2]
```diff
@@ -471,6 +472,149 @@ export const TODO3NoCustomisation = () => {
   )
 }
 
+export const TODO4WithFilterButtons = () => {
+  const [filteredLabels, setFilteredLabels] = React.useState(data.branches)
+
+  const initialSelectedLabels: string[] = ['main']
+
+  // TODO: Single selection doesn't need an array
+  const [selectedLabelIds, setSelectedLabelIds] = React.useState<string[]>(initialSelectedLabels)
```
Comment (gr2m) [inline src/drafts/SelectPanel2/SelectPanel.stories.tsx:481]: note that the way the story is implemented, selecting a tag will also select a branch

https://github.com/primer/react/assets/39992/3a0d7069-6e40-425f-8893-bead002b2ab0

Comment (siddharthkp) [reply src/drafts/SelectPanel2/SelectPanel.stories.tsx:481]: Yes, I was only focusing on the tabs bit! I can clean this story up in a follow up PR?

**Review by gr2m — APPROVED** [review]

File: src/drafts/SelectPanel2/SelectPanel.stories.tsx
Comment (broccolinisoup) [inline src/drafts/SelectPanel2/SelectPanel.stories.tsx:1]: One comment about the stories that is unrelated to this PR so feel free to address it later.
There are a11y violations you can see in the accessibility tab in the story. One is about color contrast ratio (I don't think this is related to the select panel at all) and the other one is "Scrollable region must have keyboard access".

File: src/drafts/SelectPanel2/SelectPanel.tsx [hunk 3]
```diff
@@ -116,15 +122,18 @@ const SelectPanelHeader: React.FC<React.PropsWithChildren> = ({children, ...prop
         {slots.heading}
         <Box>
           {/* Will not need tooltip after https://github.com/primer/react/issues/2008 */}
-          <Tooltip text="Clear selection" direction="s" onClick={onClearSelection}>
-            <IconButton type="button" variant="invisible" icon={ClearIcon} aria-label="Clear selection" />
-          </Tooltip>
+          {onClearSelection ? (
+            <Tooltip text="Clear selection" direction="s" onClick={onClearSelection}>
```
Comment (broccolinisoup) [inline src/drafts/SelectPanel2/SelectPanel.tsx:126]: Hopefully we can update here to use [the new Tooltip](https://github.com/primer/react/pull/3394) soon 🤞🏻

**Review by broccolinisoup — APPROVED** [review]
Left a few comments but nothing blocking! I loved reviewing this PR and reading about the accessibility notes ✨

**Comment by gr2m** [comment]
I'll make sure to keep track of your notes @broccolinisoup but will merge in the PR as I want to work on the SelectPanel to align the current implementation with our API discussion, to get it done before Sid comes back on Wednesday

**Pull request merged**